internal/pubsub/mempubsub/mem.go

```diff
@@ -37,8 +58,8 @@ type topic struct {
 
 // OpenTopic establishes a new topic.
 // Open subscribers for the topic before publishing.
-func OpenTopic() driver.Topic {
-	return &topic{}
+func OpenTopic(b *Broker, name string) *pubsub.Topic {
+	return pubsub.NewTopic(b.topic(name))
```

Comment:
So this will call `pubsub.NewTopic(nil)` if you provide a nonexistent topic name, right? How/when will this fail? (Same question for `OpenSubscription` below).

To be most similar to "real" implementations, my guess would be that `OpenTopic|Subscription` should succeed, but that reading/writing from the result always fail. Or, are we asking driver implementations to do an RPC during `Open`? Either way, the driver interface should document the expected behavior (and the conformance test should test it).

Reply:
I opened #759 to discuss.